src/wp-admin/includes/image.php

```diff
@@ -331,18 +364,73 @@ function wp_create_image_subsizes( $file, $attachment_id ) {
 
 			if ( true === $rotated ) {
 				// Append `-rotated` to the image file name.
-				$saved = $editor->save( $editor->generate_filename( 'rotated' ) );
+				$saved = $editor->save( $editor->generate_filename( 'rotated' ), $primary_mime_type );
 
 				if ( ! is_wp_error( $saved ) ) {
 					$image_meta = _wp_image_meta_replace_original( $saved, $file, $image_meta, $attachment_id );
 
+					$image_meta['sources'][ $primary_mime_type ] = _wp_get_sources_from_meta( $saved );
+
 					// Update the stored EXIF data.
 					if ( ! empty( $image_meta['image_meta']['orientation'] ) ) {
 						$image_meta['image_meta']['orientation'] = 1;
 					}
+					wp_update_attachment_metadata( $attachment_id, $image_meta );
 				} else {
 					// TODO: Log errors.
 				}
+
+				// Save additional mime types.
+				foreach ( $additional_mime_types as $output_mime_type ) {
+					$editor = wp_get_image_editor( $file );
+
+					if ( is_wp_error( $editor ) ) {
+						// This image cannot be edited.
+						continue;
+					}
+
+					if ( empty( $mime_extension_map[ $output_mime_type ] ) ) {
+						// Skip mime types that are not supported.
+						continue;
+					}
+
+					$saved = $editor->save( $editor->generate_filename( 'rotated', null, $mime_extension_map[ $output_mime_type ] ), $output_mime_type );
+					if ( ! is_wp_error( $saved ) ) {
+						$image_meta['sources'][ $output_mime_type ] = _wp_get_sources_from_meta( $saved );
+						wp_update_attachment_metadata( $attachment_id, $image_meta );
+					}
+				}
+			}
+		}
+		// Ensure presence a top level sources array for the full size images.
+		if ( ! isset( $image_meta['sources'] ) || ! is_array( $image_meta['sources'] ) ) {
+			$image_meta['sources'] = array();
+		}
```

Comment:
I believe this needs to be on top of the function? Otherwise you'll run into undefined index warnings.

Reply:
moved

Reply:
moved